cico_deploy.sh

```diff
@@ -2,69 +2,88 @@
 set -u
 set +e
 
+echo ""
+echo "#####################################"
+echo "     STARTING BUILD VALIDATION"
+echo "#####################################"
+echo ""
+
 # Source build variables
 cat jenkins-env | grep PASS > inherit-env
 . inherit-env
 . config 
 . ~/che_image_tag.env
 
+echo "VALIDATION: Validating image ${DOCKER_HUB_NAMESPACE}/che-server:${CHE_SERVER_DOCKER_IMAGE_TAG} =="
+
 # Install oc client
 yum install -y centos-release-openshift-origin
 yum install -y origin-clients
 
+OSO_USER=mloriedo
 OSO_TOKEN=$(curl -sSL -H "X-Vault-Token: ${DOCKER_HUB_PASSWORD}" -H "Content-Type: application/json" \
 	-X GET http://li546-232.members.linode.com:8200/v1/secret/os_token | cut -d\" -f18)
 if [[ -z "$OSO_TOKEN" ]]; then
-    echo "OSO token is empty, cannot proceed with verification"
+    echo "VALIDATION: Error OSO token is empty, cannot proceed with verification"
     exit 1
 fi
-OSO_USER=mloriedo
+echo "VALIDATION: Retrieved OSO token for user ${OSO_USER}"
+
+OSIO_VERSION=$(curl -sSL http://central.maven.org/maven2/io/fabric8/online/apps/che/maven-metadata.xml | grep latest | sed -e 's,.*<latest>\([^<]*\)</latest>.*,\1,g')
+echo "VALIDATION: Retrieved latest OSIO version to use for deployment: ${OSIO_VERSION}"
+
 OSO_DOMAIN=8a09.starter-us-east-2.openshiftapps.com
 OSO_API_ENDPOINT=https://api.starter-us-east-2.openshift.com
-OSIO_VERSION=$(curl -sSL http://central.maven.org/maven2/io/fabric8/online/apps/che/maven-metadata.xml | grep latest | sed -e 's,.*<latest>\([^<]*\)</latest>.*,\1,g')
 CHE_OPENSHIFT_HOSTNAME=${OSO_USER}-che.${OSO_DOMAIN}
 CHE_OPENSHIFT_PROJECT=${OSO_USER}-che
 
 oc login ${OSO_API_ENDPOINT} --token=${OSO_TOKEN}
 oc project ${CHE_OPENSHIFT_PROJECT}
-echo "Getting version of OSIO and applying template"
+
+echo "VALIDATION: Cleaning-up previous deployment"
+oc delete all -l project=che
```

Comment:
Yeah, it works. There is one more issue - when applying a new template we check whether a pod is running immediately after applying. Sometimes could happen that old pod is still running and condition thus pass on first check but then it waits on second. As a workaround I added sleep to tests as a quickest workaround. But it would be better, to check current version of deployment and expect a version +1.